core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -144,9 +144,8 @@ public Properties getKafkaProducerProperties() {
         producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
                 "org.apache.kafka.common.serialization.ByteArraySerializer");
         producerProps.put(ProducerConfig.ACKS_CONFIG, "all");
-        producerProps.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, false);
+        producerProps.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);
 
-        producerProps.put(ProducerConfig.RETRIES_CONFIG, kafkaSettings.getRetries());
```

Comment:
Side note: I think as an alternative, those (and other) configs could made configurable (and optional, so that if not set externally, they won't be set here and the Kafka client would use the default values). This way would allow to experiment with settings at deployment time wo/ rebuilding the project.

Reply:
You are right.  I'm jumping ahead of myself a bit here: the idea is to reduce the temptation to tune these parameters, once they are tested, but we haven't tested them yet.